packages/web-components/src/field/field.ts

```diff
@@ -79,41 +102,62 @@ export class Field extends FASTElement {
    *
    * @public
    */
+  @observable
   public input!: SlottableInput;
 
+  /**
+   * Updates the field's states and label properties when the assigned input changes.
+   *
+   * @param prev - the previous input
+   * @param next - the current input
+   */
+  public inputChanged(prev: SlottableInput | undefined, next: SlottableInput | undefined) {
+    if (next) {
+      this.setStates();
+      this.setLabelProperties();
+    }
+  }
+
   /**
    * Calls the `setStates` method when a `change` event is emitted from the slotted input.
    *
    * @param e - the event object
    * @internal
    */
-  public changeHandler(e: Event): void {
+  public changeHandler(e: Event): boolean | void {
     this.setStates();
+    this.setValidationStates();
+
+    return true;
   }
 
   /**
    * Redirects `click` events to the slotted input.
    *
+   * @param e - the event object
    * @internal
    */
   public clickHandler(e: MouseEvent): boolean | void {
-    if (this.isSameNode(e.target as Node | null)) {
-      this.input.focus();
+    if (this === e.target) {
       this.input.click();
-      return;
     }
 
     return true;
   }
 
+  constructor() {
+    super();
+    this.elementInternals.role = 'presentation';
```

Comment:
This is just so I understand, why do we set this in the constructor over the connectedCallback?

Reply:
The `constructor()` is only called once when the element is instantiated, whereas the `connectedCallback()` is called every time the element is attached to the DOM (which can happen any number of times).